octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+                                                                           **kwargs)):
+            return order
+        if self.get_order_using_stop_params.__name__ in defined_methods and \
+                (order := await self.get_order_using_stop_params(order_id, symbol,
+                                                                 check_completeness=check_completeness)):
+            return order
+        if self.get_trade.__name__ in defined_methods and \
+                (order := await self.get_trade(order_id, symbol, check_completeness=check_completeness)):
+            return order
+        self.logger.debug(f"Order not found using get_order: {order_id} / {symbol} - order might not exist anymore")
+
+    async def get_order_default(self, order_id: str, symbol: str = None, check_completeness: bool = True,
+                                **kwargs: dict) -> typing.Optional[dict]:
+        if self.client.has.get('fetchOrder'):
             try:
                 with self.error_describer():
                     params = kwargs.pop("params", {})
-                    return await self.client.fetch_order(order_id, symbol, params=params, **kwargs)
-                # self.exchange_manager.exchange_personal_data.upsert_order(order_id, updated_order) TODO
+                    if order := await self.client.fetch_order(order_id, symbol, params=params):
+                        await self.exchange_manager.exchange.parse_order(order, check_completeness=check_completeness)
             except ccxt.OrderNotFound:
                 # some exchanges are throwing this error when an order is cancelled (ex: coinbase pro)
                 pass
             except ccxt.NotSupported as e:
-                # some exchanges are throwing this error when an order is cancelled (ex: coinbase pro)
-                raise octobot_trading.errors.NotSupported from e
-        else:
-            # When fetch_order is not supported, uses get_open_orders and extract order id
-            open_orders = await self.get_open_orders(symbol=symbol)
-            for order in open_orders:
-                if order.get(ecoc.ID.value, None) == order_id:
-                    return order
+                self.logger.exception(e, True, "Failed to fetch order using get_order_default: Not Supported")
+            except Exception as e:
+                self.logger.exception(e, True, "Failed to fetch order using get_order_default")
+        return None
+
+    async def get_order_using_stop_params(self, order_id: str, symbol: str = None,
+                                          check_completeness: bool = True, **kwargs: dict) -> typing.Optional[dict]:
+        if self.client.has.get('fetchOrder'):
+            try:
+                with self.error_describer():
+                    params = kwargs.pop("params", {})
+                    if params := self.exchange_manager.exchange.custom_get_order_stop_params(order_id, params):
```

Comment:
Why shouldn't we call fetch_order when `if params` is False?

Reply:
this method only makes sense if you also override custom_get_order_stop_params in the tentacle
without params its the same as get_order_default so we dont need to do it again